data/src/main/java/com/linkedin/data/template/Custom.java

```diff
@@ -149,4 +151,16 @@ public static void initializeCustomClass(Class<?> customClass)
   {
     DataTemplateUtil.initializeClass(customClass);
   }
+
+  /**
+   * Return whether the input class is a Custom type or not
+   * uses coercer map to identify as each custom type must register a coercer excluding the primitive types.
+   *
+   * @param clazz is the class to check.
+   * @return true if an object of the input class is a Custom type.
+   */
+  public static boolean isCustomType(Class<?> clazz)
+  {
+    return !DataSchemaUtil.isPrimitiveClass(clazz) && DataTemplateUtil.hasCoercer(clazz);
```

Comment:
Thinking about this now. How will this behave for enums? You should test this. It looks like the `Enum` type has a coercer, yet it's not in the "primitives" mapping. Would this method return true?

Reply:
removed the changes from typespec as its anyway legacy code and not needed for fix as it handled in action request builder change.

Reply:
Did you check this wouldn't be needed for other places that use TypeSpec? I think it would still be good to have the check in TypeSpec as fallback. It doesn't hurt to be extra safe?

Reply:
will take it as a separate PR since adding multiple checks like primitive and enum didn't seem right.
Also, BatchRequests are build in a same way which would affect the performance but other places the class is constant.